xen/arch/x86/boot/slaunch_early.c

```diff
@@ -104,13 +121,43 @@ static void verify_pmr_ranges(struct txt_os_mle_data *os_mle,
 void slaunch_early_tests(uint32_t load_base_addr,
                          uint32_t tgt_base_addr,
                          uint32_t tgt_end_addr,
+                         uint32_t slaunch_param,
                          struct early_tests_results *result)
 {
     void *txt_heap;
     struct txt_os_mle_data *os_mle;
     struct txt_os_sinit_data *os_sinit;
     uint32_t size = tgt_end_addr - tgt_base_addr;
 
+    if ( !is_intel_cpu() )
+    {
+        /*
+         * Not an Intel CPU. Currently the only other option is AMD with SKINIT
+         * and secure-kernel-loader.
+         */
+        struct slr_table *slrt;
+        struct slr_entry_dl_info *dl_info;
+
+        const uint16_t *sl_header = (void *)slaunch_param;
+        /*
+         * The fourth 16-bit integer of SKL's header is an offset to
+         * bootloader's data, which is SLRT.
+         */
+        result->slrt_pa = slaunch_param + sl_header[3];
+        slrt = (struct slr_table *)result->slrt_pa;
+
+        result->mbi_pa = 0;
+        dl_info = (struct slr_entry_dl_info *)
+                  slr_next_entry_by_tag (slrt, NULL, SLR_ENTRY_DL_INFO);
+        /* Basic checks only, SKL checked and consumed the rest. */
+        if ( dl_info                             == NULL
+             || dl_info->hdr.size                != sizeof(*dl_info)
+             || dl_info->bl_context.bootloader   != SLR_BOOTLOADER_GRUB )
+            result->mbi_pa = dl_info->bl_context.context;
```

Comment:
The condition is inverted.

Reply:
Nice catch, it would be difficult to debug.
https://github.com/TrenchBoot/xen/commit/d793572382cf94d64a25a4725370ea7f7439cc35#diff-584be1f5aaa2f0679e692f6344b2e03670316e2ebdb7f4ba1dc4fbfb513d196cR156